src/Installer/dnup/DnupSharedManifest.cs

```diff
@@ -102,7 +102,7 @@ public void RemoveInstalledVersion(DotnetInstall version)
         EnsureManifestExists();
 
         var installs = GetInstalledVersions().ToList();
-        installs.RemoveAll(i => i.Id == version.Id && i.FullySpecifiedVersion == version.FullySpecifiedVersion);
+        installs.RemoveAll(i => DnupUtilities.PathsEqual(i.InstallRoot.Path, version.InstallRoot.Path) && i.Version.Equals(version.Version));
```

Comment:
I think we might actually want an ID comparison here if we want to have 'dependents' to an install where we remove the dependents and ref count. And we may have a force option that removes them like this logic does. With that said, I haven't thought that logic through 100% yet so it might change.